test/functional/lua/utility_functions_spec.lua

```diff
@@ -106,3 +108,96 @@ describe('vim.stricmp', function()
     eq(1, funcs.luaeval('vim.stricmp("\\0C\\0", "\\0B\\0")'))
   end)
 end)
+
+describe("vim.split", function()
+  it("works", function()
+    local split = function(str, sep)
+      return funcs.luaeval('vim.split(_A.s, _A.e)', { s = str, e = sep })
+    end
+
+    local tests = {
+      { "a,b", ",", false, { 'a', 'b' } },
+      { ":aa::bb:", ":", false, { '', 'aa', '', 'bb', '' } },
+      { "::ee::ff:", ":", false, { '', '', 'ee', '', 'ff', '' } },
+      { "ab", ".", false, { '', '', '' } },
+      { "a1b2c", "[0-9]", false, { 'a', 'b', 'c' } },
+      { "xy", "", false, { 'x', 'y' } },
+      { "here be dragons", " ", false, { "here", "be", "dragons"} },
+      { "axaby", "ab?", false, { '', 'x', 'y' } },
+      { "f v2v v3v w2w ", "([vw])2%1", false, { 'f ', ' v3v ', ' ' } },
+      { "x*yz*oo*l", "*", true, { 'x', 'yz', 'oo', 'l' } },
+    }
+
+    for _, t in ipairs(tests) do
+      eq(t[4], split(t[1], t[2], t[3]))
+    end
+
+    local loops = {
+      { "abc", ".-", { 'a', 'b', 'c'} },
```

Comment:
on purpose that third arg `{ 'a', 'b', 'c'}` is not used? guess it should be deleted then.